### PR TITLE
Switch to direct-to-spirv backend as default.

### LINKED ISSUE #3477: Make OptionKind::EmitSpirvDirectly turn ON by default for target CodeGenTarget target SPIRV
Blocked by:
- #3547
- Fix the perf issue exposed by #3547.
- #3440
- #3438 
- #3437

### PR DESCRIPTION
**Pull request by csyonghe**
The change is mostly about making all existing tests to pass.

Identified a couple tests where we were missing the coverage for direct to spirv backend.
They are listed in issue #4001.

Closes #3477.

Another big fix included in this PR is that our spirv emit logic for texture load/sample/store operations where the element type is `half` or `vector<half>` is incorrect. SPIRV does not allow `half` to be used as texture element type and we can only define it as `float` and convert them into `half`.

### CODE REVIEW

**Comment by ArielG-NV** [comment]
should we be adding to all tests such as `rt-pipeline-intrinsics-rgen.slang` a direct to spirv test instead of the should-be-avoided spirvViaGLSL path?

**Comment by csyonghe** [comment]
I am tracking all that in a separate issue here: https://github.com/shader-slang/slang/issues/4001

For this PR, the goal is not to lose test coverage.

**Comment by ArielG-NV** [comment]
PR looks good to me then.

File: tools/slang-test/options.cpp [hunk 1]
```diff
@@ -290,9 +290,9 @@ static bool _isSubCommand(const char* arg)
         {
             optionsOut->skipApiDetection = true;
         }
-        else if (strcmp(arg, "-emit-spirv-directly") == 0)
+        else if (strcmp(arg, "-emit-spirv-via-glsl") == 0)
         {
-            optionsOut->emitSPIRVDirectly = true;
+            optionsOut->emitSPIRVDirectly = false;
         }
```
Comment (jkwak-work) [inline tools/slang-test/options.cpp:296]: Shouldn't we still support "-emit-spirv-directly"?
"tests/compute/half_texture.slang", as an example, still uses the option.

**Review by jkwak-work — APPROVED** [review]
Looks good to me.

File: tools/slang-test/options.cpp (hunk 1 above)
Comment (csyonghe) [reply tools/slang-test/options.cpp:296]: We do, it just no longer has an effect because emit spirv directly is the default.

**Pull request merged**